### ISSUE TITLE
Allow predictions for multiple fields per model

### ISSUE THREAD

**Issue by scottpurdy** [numenta/nupic#1712]
It would be nice if you could specify multiple fields to make predictions for.

This will require a change to the CLAModel to support multiple classifier regions. It currently supports only one with the region name "Classifier":
https://github.com/numenta/nupic/blob/0cc904e5835a45e541eb0cac92901271b8f91108/nupic/frameworks/opf/clamodel.py#L1117

Additionally, the OPF and description.py format will need to be updated to support multiple fields to predict. It require some changes to metrics or swarming so it knows which field (or fields?) to use for optimization.

There are two implementations of the classifier, so we will need a task in nupic.core for the C++ implementation.

**Comment by subutai** [comment]
OPF is setup in an OO way to support different model features. Is it possible to implement this feature as a subclass of CLAModel or perhaps as a separate independent model type?  This would avoid disrupting or complicating the (already overly complex) clamodel.py.  The factory would be used to instantiate the correct class type and after that everything else should just work.

**Comment by rhyolight** [comment]
@salexashenko You going to tackle this?

**Comment by salexashenko** [comment]
Yes

**Comment by JonnoFTW** [comment]
Any progress on this so far?

**Comment by rhyolight** [comment]
Nothing yet.

**Comment by yyccR** [comment]
So, any progress on this till now?

**Comment by rhyolight** [comment]
Nothin yet. To be clear, Numenta has not prioritized this task. If any work is going to get done here, it is going to be community-driven.

**Comment by barnettjv** [comment]
Guys, its 2020. Has anyone made any progress on multi field predictions? I'm hoping so.

**Comment by breznak** [comment]
> Guys, its 2020. Has anyone made any progress on multi field predictions? I'm hoping so. 

Have a look at actively developed 
https://github.com/htm-community/htm.core

We don't use OPF anymore but it's easy to setup multiple `Classifiers` to give you `multi-label` predictions. Cheers

**Comment by barnettjv** [comment]
Thank you so much!!!! My Ph D. dissertation now stands a chance!
s/
Jeff Barnett

**Comment by barnettjv** [comment]
@breznak So I finally got around to finishing my research (well nearly). I wanted to ask you again about multi-label predictions. So I've been using the htm.core posted on the htm-community github page that you referred me to for my research. I don't use the "Classifier" class/function but rather the example hotgym.py code provided as the baseline for my htm(s). So my question is regarding predictor.learn() function. I have 6 features that I'm trying to learn (actually the flag bits from EFLAGS). Each flag (0,1) scalar value is what I'm trying to predict. I currently have created a predictor for each flag (e.g., CF_Predictor, PF_Predictor, ...). Is this the correct way to go about learning and predicting the 6 features? or do I just use one predictor and send a list containing the 6 flag values? If the later, how do you pull the individual predictions from the predictions array?

**Comment by barnettjv** [comment]
Yes I was able to complete my research, however I can't remember if I had
multi-variable predictions or just ran single predictions many times. I'll
dig up my work and get back to you.
s/
Dr. Barnett
